Geometry_Engine/Compute/DistributeOutlines.cs

```diff
@@ -43,15 +43,33 @@ public static partial class Compute
         [Output("distributedOutlines", "Lists representing each set of outlines. For each list, the first outline is the containing outline and the remainder are the outlines it contains.")]
         public static List<List<Polyline>> DistributeOutlines(this List<Polyline> outlines, double tolerance = Tolerance.Distance)
         {
-            if (outlines == null || outlines.Count == 0)
+            if (outlines == null)
             {
-                BH.Engine.Reflection.Compute.RecordError("Cannot distribute a null or empty collection of outlines.");
+                BH.Engine.Reflection.Compute.RecordError("Cannot distribute a null collection of outlines.");
                 return new List<List<Polyline>>();
             }
 
-            if (outlines.Any(p => !p.IIsClosed(tolerance)))
+            if (outlines.Any(x => x == null))
+            {
+                BH.Engine.Reflection.Compute.RecordError("At least one of the input outlines is a null curve.");
+                return new List<List<Polyline>>();
+            }
+
+            if (outlines.Any(p => !p.IsClosed(tolerance)))
             {
-                BH.Engine.Reflection.Compute.RecordError("All outlines need to be closed to create a panel.");
+                BH.Engine.Reflection.Compute.RecordError("At least one of the input outlines is not closed within the input tolerance. Please try changing the tolerance if this behaviour is not expected.");
+                return new List<List<Polyline>>();
+            }
+
+            if (outlines.Any(p => !p.IsPlanar(tolerance)))
+            {
+                BH.Engine.Reflection.Compute.RecordError("At least one of the input outlines is not planar within the input tolerance. Please try changing the tolerance if this behaviour is not expected.");
+                return new List<List<Polyline>>();
+            }
+
+            if (outlines.Any(p => !p.IsSelfIntersecting(tolerance)))
```

Comment:
This causes issues as self intersecting does not use the same tolerance, in that a higher tolerance here means a curve can be self intersecting when it is actually further from self intersecting when the tolerance is higher. This is inherently problematic as a higher tolerance is often needed for the closed checks and the like with very small geometry, where the geometry is much more likely to come closer to itself as well (for example, my test file from /BHoM/Revit_Toolkit/pull/1082 fails for this reason). I suggest we either leave the tolerance on this check at the default value, or don't run this check as part of distribute outlines as before.

Reply:
Very valid point @enarhi! Actually, this highlights a wider issue of how the tolerance affects the results of different checks. I see two behaviours, depending on the tolerance gradient:
- higher tolerance => easier to pass (`IsClosed`, `IsPlanar`)
- higher tolerance => harder to pass (`IsSelfIntersecting`)

At the first sight, the trends above should be consistent across all preliminary checks, so could possibly be captured in a form of a table, what do you think @al-fisher @IsakNaslundBh? Following that, we could consider prototyping `IsValidInput` method or similar, with a range of Boolean parameters or or enum flags, in order to centralise the checking procedure for geometry inputs.

Back to the origin: I see no safe/smart solution with regard to the problem highlighted by @enarhi, therefore I simply removed the self-intersection checks from this PR - better safe than sorry in the last sprint 😉